daemon/import.go

```diff
@@ -59,15 +61,24 @@ func (daemon *Daemon) ImportImage(src string, repository, tag string, msg string
 		rc = inConfig
 	} else {
 		inConfig.Close()
+		if len(strings.Split(src, "://")) == 1 {
+			src = "http://" + src
+		}
 		u, err := url.Parse(src)
 		if err != nil {
 			return err
 		}
-		if u.Scheme == "" {
-			u.Scheme = "http"
-			u.Host = src
-			u.Path = ""
+
+		// Checking the file referenced by the URL exists to avoid sending
+		// status code 200 to the client although the download process will fail
+		resp, err = http.Head(u.String())
```

Comment:
Although bad practice, `HEAD` requests can be blocked / not supported. In that case, I think standard practice is that the client should fall back to a `GET`. Doing a `GET` here most likely won't make sense though